lib/os/Kconfig

```diff
@@ -46,16 +46,6 @@ config SYS_HEAP_ALLOC_LOOPS
 	  keeps the maximum runtime at a tight bound so that the heap
 	  is useful in locked or ISR contexts.
 
-config SYS_HEAP_ALIGNED_ALLOC
-	bool "Enable sys_heap_aligned_alloc() API"
```

Comment:
The explanation in the commit message is good, but it would be best to repeat that in a comment in mem_to_chunkid() to explain things.  That this rule works is subtle and surprising.